### DISCUSSION TITLE
Is it possible to have two transactions back to back?

### DISCUSSION THREAD

**Discussion by builtbyleo** [wevm/wagmi#1033] [Q&A]
Im trying to get two transactions to occur back to back . The way i have it now seems to work every 4/5 tries but I cant get it work consistently. Essentially in my react app i have a button that calls:
  
    const handleStake = async () => {
        await approveTx.writeAsync?.().then(async () => 
        await fundMeTx.writeAsync?.() 
        )
        
    }

These refer to the following transactions: 

    const prepareFundMe = usePrepareContractWrite({
        addressOrName: proxyAddress,
        contractInterface: voteProxyABI,
        functionName: 'fundMe',
        args: [ Web3.utils.toWei(stakeAmount, 'ether'), Web3.utils.hexToNumber(proposalID), tsryPrice],
        chainId: 5,
      })

```
const prepareApprove = usePrepareContractWrite({
            addressOrName: treasuryTokenAddress,
            contractInterface: tsrytokenABI,
            functionName: 'approve',
            args: [proxyAddress, Web3.utils.toWei(stakeAmount, 'ether')],
            chainId: 5,
        })
```

  
    const fundMeTx = useContractWrite(prepareFundMe.config)
    const approveTx = useContractWrite(prepareApprove.config)

The result im trying to achieve is that if a user clicks on that button, the metamask wallet will appear, first asking the user to approve the the transferFrom, then immediately ask the user to confirm the 'fundMe' transaction. (metamask is my wallet)

any help would be appreciated

**Comment by SpasiboKojima** [comment]
There's no need for `.then()` in your code, you can just use 
```
const handleStake = async () => {
    await approveTx.writeAsync?.()
    await fundMeTx.writeAsync?.() 
}
```
What's the issue you are facing 1/5 times? Transaction revert, out of gas?
